app/controllers/submissions_controller.rb

```diff
@@ -874,4 +1001,12 @@ def is_valid_url?(url)
   rescue URI::InvalidURIError
     false
   end
+
+  def select_file
+    params[:select_file_id] && SubmissionFile.find_by(id: params[:select_file_id])
```

Comment:
Ah, we can now (more nicely) do `record.submission_files.find_by`, so that we're only looking for the submission files of the current submission